patches/server/0728-Add-Recipe-API.patch

```diff
@@ -0,0 +1,183 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: lexikiq <noellekiq@gmail.com>
+Date: Thu, 15 Jul 2021 01:39:12 -0400
+Subject: [PATCH] Add Recipe API
+
+Adds methods to utilize a world's RecipeManager to obtain recipes
+
+diff --git a/src/main/java/io/papermc/paper/inventory/CraftRecipeType.java b/src/main/java/io/papermc/paper/inventory/CraftRecipeType.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..890d5e3ffd5da23f05d456232568fc6d7e07b214
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/CraftRecipeType.java
+@@ -0,0 +1,15 @@
++package io.papermc.paper.inventory;
++
++import net.minecraft.core.Registry;
++import org.bukkit.NamespacedKey;
++import org.bukkit.craftbukkit.util.CraftNamespacedKey;
++
++public class CraftRecipeType {
++    public static net.minecraft.world.item.crafting.RecipeType asNMS(RecipeType bukkit) {
++        return Registry.RECIPE_TYPE.get(CraftNamespacedKey.toMinecraft(bukkit.getKey()));
++    }
++
++    public static RecipeType asBukkit(net.minecraft.world.item.crafting.RecipeType nms) {
++        return RecipeType.getByKey(NamespacedKey.minecraft(nms.toString()));
```

Comment:
I'd use the `Registry.RECIPE_TYPE` to do this conversion as well. It has methods to get the `ResourceLocation` from the object. Its a two-way map.